spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala

```diff
@@ -101,6 +101,20 @@ class BigQueryStrategy(expressionConverter: SparkExpressionConverter, expression
     Some(sourceQuery.bigQueryRDDFactory)
   }
 
+  /** Removes project plan with no fields and returns the child, to continue the generation of query.
+   * Spark is generating an empty project plan when the query has count(*)
+   * Found this issue when executing query-9 of TPC-DS
+   *
+   * @param plan
+   * @return
+   */
+  def cleanUpLogicalPlan(plan: LogicalPlan): LogicalPlan = {
```

Comment:
We should use plan.transform here and recurse down the tree before starting the query generation process so that we have a clean tree

```
def cleanUpLogicalPlan(plan: LogicalPlan): LogicalPlan = {
    plan.transform({
      case Project(Nil, child) => child
    })
```